geomstats/_backend/tensorflow/__init__.py

```diff
@@ -553,6 +554,39 @@ def cast(x, dtype):
     return tf.cast(x, dtype)
 
 
+def broadcast_arrays(x, y, **kwargs):
+    tensors = [x, y]
+    shapes = [t.get_shape().as_list() for t in tensors]
+    max_rank = max(len(s) for s in shapes)
+
+    for index, value in enumerate(shapes):
+        shape = value
+        if len(shape) == max_rank:
+            continue
+
+        tensor = tensors[index]
```

Comment:
These lines are not tested.

Reply:
Added test_broadcast_arrays and error tests as well

Reply:
> Very nice! 🎉 Good to go when the remaining unit tests are added!
> 
> About the DeepSource error: it is fine to have that error from a test file: we do want to test the private functions. So if you prefer having `_dist` and the error only comes from the test file, that is fine! However, `dist_broadcast` is a good explicit name: maybe it's even better to keep it like this?

Yes dist_broadcast is kept